### PR TITLE
extract field method to request-base

### PR DESCRIPTION
**Pull request by focusaurus**
deal with root/globals
document both browser and node "val" param types

part of #830

### CODE REVIEW

File: lib/request-base.js [hunk 1]
```diff
@@ -3,6 +3,19 @@
...
+ * Root reference for environment-specific globals.
+ */
+
+var root;
+if (typeof window !== 'undefined') { // Browser window
+  root = window;
+} else if (typeof self !== 'undefined') { // Web Worker
+  root = self;
+} else { // Other environments (node or other)
+  root = {FormData: require('form-data')};
```
Comment (pornel) [inline lib/request-base.js]: How does it behave when compiled for the browser? Will it cause an unused module to be bundled?

Comment (focusaurus) [inline lib/request-base.js]: Good point. I checked and it looks like form-data is properly browserified to use the built-in `FormData` on the browser:

https://github.com/form-data/form-data/blob/master/lib/browser.js (their package.json file has the proper "browser" property as well).

I'm OK with this approach overall in dealing with node/browser discrepancies. We'll have a few others to address soon enough like btoa, Buffer, etc.

Comment (focusaurus) [inline lib/request-base.js]: And actually, based on this it's probably OK to just do a direct `var FormData = require('form-data');` and let browserify do it's thing.

**Comment by pornel** [comment]
neat

**Pull request merged**